optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/chained/ChainedSwapMove.java

```diff
@@ -74,8 +77,8 @@ protected void doMoveOnGenuineVariables(ScoreDirector<Solution_> scoreDirector)
                     scoreDirector.changeVariableFacade(variableDescriptor, leftEntity, oldRightValue);
                     scoreDirector.changeVariableFacade(variableDescriptor, rightEntity, oldLeftValue);
                 } else {
-                    Object oldLeftTrailingEntity = oldLeftTrailingEntityList.get(i);
-                    Object oldRightTrailingEntity = oldRightTrailingEntityList.get(i);
+                    Object oldLeftTrailingEntity = oldLeftTrailingEntityMap.get(variableDescriptor);
+                    Object oldRightTrailingEntity = oldRightTrailingEntityMap.get(variableDescriptor);
```

Comment:
So when I have a plannning entity `LessonVisit` with a chained planning variable `previous` and a non-chained planning variable `room`, this code will execute for the `room`  too? That `room` should behave like a normal swap. Does it?

What happens to `room` if it's trailing entities are null, both left and right? I guess it behaves like a normal swap move indeed?

Reply:
I think the code behaves correctly, but we should add a test for it.